### DISCUSSION TITLE
🚀  Required fields should not add a * automatically

### DISCUSSION THREAD

**Discussion by fraincs** [gsoft-inc/sg-orbit#1127] [Ideas]
## Description

Enforcing a * on required field labels has been problematic in instances where all fields or most fields are required. In certain instances there is a need to write required instead of having a * or not when a form has a single field a required * seems unnecessary.

That's why we should let it to the user to add the required / * when needed.

**Comment by fraincs** [comment]
Upon further thinking adding a prop in order to add a required suffix to the labels when needed, an optional suffix should also be available.

necessityIndicator could be a prop available at the Form level as well as at the Field level, adding * to required fields when necessityIndicator is set to `required` and (optional) to non required fields when necessityIndicator is set to `optional`.

**Comment by patricklafrance** [comment]
I am not sure letting the developer choose it's own "required / *" is a good idea since Orbit is a design system and should provide a consistant way to indicate required fields.

**Comment by fraincs** [comment]
We're still stuck with an issue where in some instances a required is better, like in small forme (signup forms for exemple) and redundant when all fields are required ou all except one.

**Comment by patricklafrance** [comment]
What is the issue when all fields are required? Or all except one?

The fact that the developer must specify the `required` prop on all the `Label` components? Is so, could you elaborate on why it's an issue? (other than it's redundant)

I don't get why for some forms the necessity indicator should be different? Do you have any examples that could help me understand? My thinking is that Orbit purpose is to build a coherent system for our UI. With that in mind, I don't get why we'll want to show our users different necessity indicators?
